Form/JQuery/Type/Select2Type.php

```diff
@@ -89,13 +90,16 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function getParent()
     {
-        return $this->widget;
+        if (class_exists('Symfony\Component\Form\Extension\Core\Type\\' . ucfirst($this->widget) . 'Type')) {
+            return 'Symfony\Component\Form\Extension\Core\Type\\' . ucfirst($this->widget) . 'Type';
+        }
+        return 'Symfony\Bridge\Doctrine\Form\Type\\' . ucfirst($this->widget) . 'Type';
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getName()
+    public function getBlockPrefix()
```

Comment:
As of 3.0 you can only pass the class, it the form type needs an object it can be registered as a service.

I'll merge your PR for now, removing some deprecations doesn't hurt. But please look for alternatives, I'll probably deprecate this whole feature as there seems to be no simple fix for Symfony 3. If you want to "selectify" a choice type, you can just add a css class to it and have some js around.